### PR TITLE
chore(react-sdk): Prepare for 1.2.0-alpha.1 release

### PR DESCRIPTION
**Pull request by circAssimilate**
## Summary
- Updated minor version of core SDK (`@optimizely/optimizely-sdk`) dependency which, for unrecognized events sent via `.track()`, will impact the log levels in this SDK as well.
- Bumps the version to `1.2.0-alpha.1`
- Updates changelog

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,24 +1,32 @@
 # Changelog
+
 All notable changes to this project will be documented in this file.
 
 The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
 
+## [1.2.0 alpha 1] - March 4th, 2020
```
Comment (jamesopti) [inline CHANGELOG.md]: Nit: `1.2.0-alpha.1`

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,24 +1,32 @@
...
+
+### New Features
+
+- Updated minor version of core `@optimizely/optimizely-sdk` dependency
```
Comment (mjc1283) [inline CHANGELOG.md]: Let's mention the effect of this: the change in the core SDK which will impact the log levels in this SDK too.

File: CHANGELOG.md (hunk 1 above)
Comment (circAssimilate) [reply CHANGELOG.md]: I actually had that at first but needed to change it due to the lint spell checker! @mjc1283, any suggestions on overriding that for this line?

Comment (circAssimilate) [reply CHANGELOG.md]: Looks like it's something from `markdown-spellcheck` but I'm not sure how to override it (or if it's worth doing so).

Comment (circAssimilate) [reply CHANGELOG.md]: Tried to add an entry to `.spelling` which I got to work in a local project.

Comment (circAssimilate) [reply CHANGELOG.md]: Got it with @mjc1283 and @juancarlostong's help. Just had to push new version to https://github.com/optimizely/mdspell-config (and JC is aiming to improve this)

**Pull request merged**